docs/migration-to-NRI.md

```diff
@@ -0,0 +1,90 @@
+ - systemctl status cri-resource-manager
+
+## 3. Run: "systemctl cat kubelet". Look for: EnvironmentFile=/.../kubeadm-flags.env
+
+## 4. Run: "sudo vim /.../kubeadm-flags.env"
+
+## 5. Change the --container-runtime-endpoint from containerd socket to cri-resmgr.scok:
+ - KUBELET_KUBEADM_ARGS="--container-runtime-endpoint=unix:///var/run/cri-resmgr/cri-resmgr.sock --pod-infra-container-image=registry.k8s.io/pause:3.9"
+
+ Alternatively run:
+ - kubelet <other-kubelet-options> --container-runtime=remote \
+     --container-runtime-endpoint=unix:///var/run/cri-resmgr/cri-resmgr.sock
+
+## 6. Restart kubelet: "systemctl restart kubelet"
+
+## 7. Run CRI-RM with your desired policy with the following command:
+ - cri-resmgr --force-config <config-file> --runtime-socket unix:///var/run/containerd/containerd.sock
+
+## 8. Deploy your pod
+
+## 9. See what resources the container got assigned with:
+ - kubectl exec -c $container $pod -- cat /proc/self/status | grep _allowed_list
+
+Output should be something along the lines of:
+
+Cpus_allowed_list:	10-15 </br>
+Mems_allowed_list:	1
+
+# Enabling NRI
+
+## 1. With Containerd as the runtime:
+
+Replace containerd in the system with 1.7 or newer version (NRI server not supported in older versions).
+
+Change the Kubelet --container-runtime-endpoint to containerd socket:
+ - sudo vim /.../kubeadm-flags.env
+ - KUBELET_KUBEADM_ARGS="--container-runtime-endpoint=unix:///var/run/containerd/containerd.sock --pod-infra-container-image=registry.k8s.io/pause:3.9"
+ - sudo vim /etc/sysconfig/kubelet 
+ - KUBELET_EXTRA_ARGS= --container-runtime-endpoint=/var/run/containerd/containerd.sock <- Remember this aswell
+ - systemctl restart containerd
+ - systemctl restart kubelet
+
+Edit the containerd config file and look for the section "io.containerd.nri.v1.nri" and replace "disable = true" with "disable = false":
+ - "vim /etc/containerd/config.toml"
+
+## 2. Build the policies:
+ - git clone https://github.com/jukkar/nri-resmgr.git
+ - cd nri-resmgr
+ - make
+ - make images
+
+## 3. Apply required crds:
+ - kubectl apply -f deployment/base/crds/noderesourcetopology_crd.yaml
+
+## 4. Start the NRI plugin you want to run:
+ - ctr -n k8s.io images import build/images/nri-resmgr-topology-aware-image-9797e8de7107.tar
+ - kubectl apply -f build/images/nri-resmgr-topology-aware-deployment.yaml
+
+## 5. Deploy your pod.
+
+## 6. See the resources the pod got assigned with:
+ - kubectl exec $pod -c $container  -- grep allowed_list: /proc/self/status
+ - Output should look similar to the output of CRI-RM
+
+## Common error:
+NRI socket errors, ex: "failed to start resource manager: failed to start NRI plugin: failed to connect to NRI service: dial unix /var/run/nri.sock: connect: connection refused"
+ - sudo vim /etc/containerd/config.toml
+ - Go to the "io.containerd.nri.v1.nri" and look for "socket_path"
+ - Make sure this matches the socket path of your NRI socket.
+ - systemctl restart containerd
```

Comment:
This error should not occur if the NRI versions used to compile the runtime and the policies are in sync (or at least both are past the version where the default socket path change happened, past commit d4f68f05d880). Considering that, I think this would probably come through nicer, if we point them to use
- at least the freshly released containerd 1.7 (which uses NRI v0.3.0).
- at least the freshly released CRI-O 1.26.2 (which uses NRI v0.3.0).
- to use our 'stable' branch (@jukkar maybe the 'preview' one which then needs updating) which uses at least NRI v0.3.0